### PR TITLE
Add some key version 25 capabilitles

### PR DESCRIPTION
**Pull request by david-a-wheeler**
Update documentation by noting some of the version 25 capabilities. This doesn't add everything in version 25, but I wanted to at least get *started* on updating this document.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -3992,26 +3998,68 @@ The search pattern language is very simple,
...
 
-* A constant pattern only matches the same constant.
-* A typecode pattern matches any variable with that typecode.
+~~~~bnf
+search-pattern ::= sub-pattern*
+sub-pattern ::= ['$'modifiers] MATH-SYMBOL*
+modifiers ::= modifier*
+modifier := '+' | '!' | 'h' | 'a'
+~~~~
+
```
Comment (expln) [inline docs/index.md:4010]: It was a mistake in my documentation. I updated it. A search pattern must be a space-separated sequence of **one** or
more sub-patterns. Updated grammar:
```
search-pattern ::= sub-pattern+
sub-pattern ::= ['$'modifiers] MATH-SYMBOL+
modifiers ::= modifier+
modifier := '+' | '!' | 'h' | 'a'
```

**Review by expln — APPROVED** [review]

**Comment by expln** [comment]
> This doesn't add everything in version 25, but I wanted to at least get started on updating this document.

I don't think it is needed to add everything. Only some main features would be enough. And looks like the search by pattern is the only one which should be added because it adds a lot of flexibility in the search.

**Comment by david-a-wheeler** [comment]
I think the bookmarks, at least, should also be noted in the documentation :-).

**Comment by david-a-wheeler** [comment]
How about:

~~~~bnf
search-pattern ::= ['$'modifiers] MATH-SYMBOL+ ( '$'modifiers MATH-SYMBOL+ )*
~~~~

**Comment by expln** [comment]
> search-pattern ::= ['$'modifiers] MATH-SYMBOL+ ( '$'modifiers MATH-SYMBOL+ )*

This also doesn't work. For example, it allows the first sub-pattern not to have modifiers in all cases. But actually a sub-pattern may not have modifiers when it is the only sub-pattern in the pattern. If there are two or more sub-patterns, then each of them must have at least one modifier `h` or `a`. For example, for the pattern `ph -> $a ps <->` you will get the error "Malformed pattern text: Each sub-pattern must be either a hypothesis or an assertion."

**Comment by expln** [comment]
Another problem is that it is not allowed to use `a` and `h` for the same sub-pattern. But this doesn't follow from the EBNF as well. I don't think this EBNF should be very precise. Currently it gives a general idea. If users misspell the pattern they will see some meaningful error message. It also should be common logic that you cannot put two sub-patterns without any modifiers because it is not clear where one ends and another begins.

File: docs/index.md (hunk 1 above)
Comment (david-a-wheeler) [reply docs/index.md:4010]: Ok, I'll use that instead!

**Pull request merged**